sql-plugin/src/main/scala/com/nvidia/spark/rapids/ParquetSchemaUtils.scala

```diff
@@ -360,18 +360,19 @@ object ParquetSchemaUtils extends Arm {
 
   private def clipSparkArrayType(
       sparkType: ArrayType,
-      parquetList: GroupType,
+      parquetList: Type,
       caseSensitive: Boolean,
       useFieldId: Boolean): DataType = {
     val elementType = sparkType.elementType
     // Unannotated repeated group should be interpreted as required list of required element, so
     // list element type is just the group itself.
     // TODO: When we drop Spark 3.1.x, this should use Parquet's LogicalTypeAnnotation
     //if (parquetList.getLogicalTypeAnnotation == null &&
-    if (parquetList.getOriginalType == null &&
+    val newSparkType = if (parquetList.getOriginalType == null &&
```

Comment:
nit: A comment here about how OriginalType is not null for leaf nodes would be nice.

Reply:
done